### PR TITLE
[RA2 Ch4]: Update of the Kubernetes Distributions & Platforms table

### LINKED ISSUE #2381: [RA2 Ch4] Kubernetes distribution table in ra2.k8s.001 is obsoleted
The [Kubernetes Distributions & Platforms table](https://docs.google.com/spreadsheets/d/1LxSqBzjOxfGx3cmtZ4EbB_BGCxT_wlxW_xgHVVa23es/edit#gid=0) in [Chapter 4.3](https://cntt.readthedocs.io/en/latest/ref_arch/kubernetes/chapters/chapter04.html#kubernetes) is obsoleted. We should figure out what is its replacement and add a reference to that.

### PR DESCRIPTION
**Pull request by CsatariGergely**
as the previous one was deprecated.

Closes #2381

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/chapter04.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ In order for the Kubernetes components to be conformant with the Reference Archi
...
 |`ra2.k8s.003`|Highly available control plane|An implementation must consist of at least one master node per availability zone or fault domain to ensure the high availability and resilience of the Kubernetes control plane services. A master node must run at least the following Kubernetes control plane services: `kube-apiserver`, `kube-scheduler` and `kube-controller-manager`|[req.gen.rsl.02](./chapter02.md#23-kubernetes-architecture-requirements)<br>[req.gen.avl.01](./chapter02.md#23-kubernetes-architecture-requirements)|
 |`ra2.k8s.012`|Highly available control plane|A master node must run at least the following Kubernetes control plane services: `kube-apiserver`, `kube-scheduler` and `kube-controller-manager`|[req.gen.rsl.02](./chapter02.md#23-kubernetes-architecture-requirements)<br>[req.gen.avl.01](./chapter02.md#23-kubernetes-architecture-requirements)|
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter04.md:78]: This is duplicate of ra2.k8s.003 above

**Review by petorre — APPROVED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter04.md (hunk 1 above)
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter04.md:78]: @pgoyal01 I've created #2394 to address this.

**Review by tomkivlin — APPROVED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter04.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ In order for the Kubernetes components to be conformant with the Reference Archi
...
 |`ra2.k8s.002`|Highly available etcd|An implementation must consist of either three, five or seven nodes running the etcd service (can be colocated on the master nodes, or can run on separate nodes, but not on worker nodes).|[req.gen.rsl.02 req.gen.avl.01](./chapter02.md#23-kubernetes-architecture-requirements)|["For the high availability requirement"](../../../ref_impl/cntt-ri2/chapters/chapter04.md#431-installation-on-bare-metal-infratructure)|
 |`ra2.k8s.003`|Highly available control plane|An implementation must consist of at least one master node per availability zone or fault domain to ensure the high availability and resilience of the Kubernetes control plane services. A master node must run at least the following Kubernetes control plane services: `kube-apiserver`, `kube-scheduler` and `kube-controller-manager`|[req.gen.rsl.02](./chapter02.md#23-kubernetes-architecture-requirements)<br>[req.gen.avl.01](./chapter02.md#23-kubernetes-architecture-requirements)|
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter04.md:77]: @tomkivlin The following is still duplicate of `ra2.k8s.012`:
"A master node must run at least the following Kubernetes control plane services: `kube-apiserver`, `kube-scheduler` and `kube-controller-manager`|[req.gen.rsl.02](./chapter02.md#23-kubernetes-architecture-requirements)<br>[req.gen.avl.01](./chapter02.md#23-kubernetes-architecture-requirements)."

Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter04.md:77]: Oh yeah, I missed that - will update, thanks!

**Pull request merged**